services/api/src/typeDefs.js

```diff
@@ -847,6 +878,32 @@ const typeDefs = gql`
     id: Int!
   }
 
+  input DeployEnvironmentLatestInput {
+    environment: EnvironmentInput!
+  }
+
+  input DeployEnvironmentBranchInput {
+    project: ProjectInput!
+    branchName: String!
+    branchRef: String
+  }
+
+  input DeployEnvironmentPullrequestInput {
+    project: ProjectInput!
+    number: Int!
+    title: String!
+    baseBranchName: String!
+    baseBranchRef: String!
+    headBranchName: String!
+    headBranchRef: String!
+  }
+
+  input DeployEnvironmentPromoteInput {
+    sourceEnvironment: EnvironmentInput!
+    project: ProjectInput!
+    branchName: String!
```

Comment:
let's call this `destinationEnvironment` and not `branchName`. Calling it `branchName` was a weird idea when I created the Promote System.

The idea is basically that with a promotion you can take any Docker Images and Services from one Environment and create a second Environment.

so it makes more sense to me to have a `sourceEnvironment` and a `destinationEnvironment` defined

Reply:
Done!